subprojects/files/src/main/java/org/gradle/api/internal/file/pattern/PatternMatcher.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.file.pattern;
```

Comment:
How about we use `org.gradle.internal.file.pattern` as package? I don't know why `api` would/should be part of the package.

Reply:
:+1: I wanted to do this but not before getting the rest reviewed.

Pattern matching code is now in a `org.gradle.internal.file.pattern`.